tree/ntuple/v7/inc/ROOT/RMiniFile.hxx

```diff
@@ -0,0 +1,221 @@
+/// \file ROOT/RMiniFile.hxx
+/// \ingroup NTuple ROOT7
+/// \author Jakob Blomer <jblomer@cern.ch>
+/// \date 2019-12-22
+/// \warning This is part of the ROOT 7 prototype! It will change without notice. It might trigger earthquakes. Feedback
+/// is welcome!
+
+/*************************************************************************
+ * Copyright (C) 1995-2019, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT7_RMiniFile
+#define ROOT7_RMiniFile
+
+#include <ROOT/RNTupleOptions.hxx>
+#include <ROOT/RStringView.hxx>
+
+#include <cstdint>
+#include <cstdio>
+#include <memory>
+#include <string>
+
+class TFile;
+
+namespace ROOT {
+namespace Experimental {
+
+namespace Detail {
+class RRawFile;
+}
+
+// clang-format off
+/**
+\class ROOT::Experimental::RNTuple
+\ingroup NTuple
+\brief Entry point for an RNTuple in a ROOT file
+
+The class points to the header and footer keys, which in turn have the references to the pages.
+Only the RNTuple key will be listed in the list of keys. Like TBaskets, the pages are "invisible" keys.
+Byte offset references in the RNTuple header and footer reference directly the data part of page records,
+skipping the TFile key part.
+*/
+// clang-format on
+struct RNTuple {
+   /// Allows for evolving the struct in future versions
+   std::uint32_t fVersion = 0;
```

Comment:
As a side note, I think we still have to upgrade TStreamerInfo (and possibly TDataMember) to correctly store those as 32 bits on all platforms (on platforms where long is 32bits and that typedef happens to use long then it will be stored as 64bits).